bundle/edu.gemini.sp.vcs/src/main/scala/edu/gemini/sp/vcs2/VcsServer.scala

```diff
@@ -109,7 +107,10 @@ class VcsServer(odb: IDBDatabaseService) { vs =>
       vs.read(id, user)(_.getVersions).unsafeRun
 
     override def add(p: ISPProgram): TryVcs[Unit] =
-      vs.add(p, user).unsafeRun
+      (for {
+        id <- (Option(p.getProgramID) \/> MissingId).liftVcs
+        _  <- accessControlled(id, user) { vs.add(p) }
+      } yield ()).unsafeRun
```

Comment:
This is the entry point in the server for committing a new file to a remote database.